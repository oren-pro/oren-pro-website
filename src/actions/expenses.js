import uuid from'uuid';
import database from '../firebase/firebase';


// this is an action generator - without database
// a component calls this action generator
// the action generator returns an object to the component
// the com ponent dispaches the object to redux
// redux store changes

// this is an action generator - with database
// a component calls this action generator
// the action generator returns a function to the component
// the com ponent dispaches the function to redux
// the function runs (has the ability to dispatch other actions to redux or do whatever it wants)
// redux store changes



// ADD_EXPENSE

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        const {
            description = '',
            notes = '',
            amount = 0,
            createdAt = 0
        } = expenseData;
        const expense = {description, notes, amount, createdAt};
        return database.ref(`users/${uid}/expenses`).push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        });
    };
};

// REMOVE_EXPENSE

export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const startRemoveExpense = ({ id } = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/expenses/${id}`).remove().then(() => {
            dispatch(removeExpense({ id }));
        })
    };
};

// // database.ref('isSingle').remove().then(() => {
// //     console.log('Removed isSingle');
// // }).catch((e) => {
// //     console.log('Remove failed: ', e);
// // });

// EDIT_EXPENSE

export const editExpense = ( id, updates ) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

export const startEditExpense = ( id, updates ) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/expenses/${id}`).update(updates).then(() => {
            dispatch(editExpense( id, updates ));
        })
    };
};

//SET_EXPENSES
export const setExpenses = (expenses) => ({
    type: "SET_EXPENSES",
    expenses
});

export const startSetExpenses = () => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/expenses`).once('value').then((snapshot) => {
            const expenses = [];
            snapshot.forEach((childSnapshot) => {
                expenses.push({
                    id: childSnapshot.key,
                    ...childSnapshot.val()
                });
            });
            dispatch(setExpenses(expenses));
        });
    };
};