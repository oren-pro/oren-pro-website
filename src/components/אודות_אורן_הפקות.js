import React from 'react';
import { connect } from 'react-redux';

export const AboutUs = () => (
    <div>
        <p>נעים להכיר</p>
    </div>
);

export default AboutUs;