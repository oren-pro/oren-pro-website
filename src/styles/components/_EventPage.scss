
.event__sharestrip__box {
    width: 68rem;
    display: block;
    margin-left: auto;
    margin-right: auto;
}


$iconWidth: 2.4rem;
$iconHeight: 2.4rem;
$iconMarginTop: 0.4rem;
$iconMarginBottom: 0.6rem;

.events__eventshare__phone {
    width: $iconWidth;
    height: $iconHeight;
    background-image: url('/images/eventspage/events-sharestrip-phone.svg');
    background-repeat: no-repeat;
    background-size: $iconWidth $iconHeight;
    float: left;
    margin-top: $iconMarginTop;
    margin-bottom: $iconMarginBottom;
    margin-right: 0.5rem;
}

.events__eventshare__phone:hover {
    background-image: url('/images/eventspage/events-sharestrip-phone-hover.svg');
}

.events__eventshare__mail {
    width: $iconWidth;
    height: $iconHeight;
    background-image: url('/images/eventspage/events-sharestrip-mail.svg');
    background-repeat: no-repeat;
    background-size: $iconWidth $iconHeight;
    float: left;
    margin-top: $iconMarginTop;
    margin-bottom: $iconMarginBottom;
    margin-right: 0.5rem;
}

.events__eventshare__mail:hover {
    background-image: url('/images/eventspage/events-sharestrip-mail-hover.svg');
}

.events__eventshare__facebook {
    width: $iconWidth;
    height: $iconHeight;
    background-image: url('/images/eventspage/events-sharestrip-facebook.svg');
    background-repeat: no-repeat;
    background-size: $iconWidth $iconHeight;
    float: left;
    margin-top: $iconMarginTop;
    margin-bottom: $iconMarginBottom;
    margin-right: 0.7rem;
}

.events__eventshare__facebook:hover {
    background-image: url('/images/eventspage/events-sharestrip-facebook-hover.svg');
}

.events__eventshare__text {
    align-self: center;
    padding-top: 1rem;
    margin: 0;
    color: $gray;
    font-size: 1.3rem;
    line-height: 1.3rem;
    float: left;
}

.events__eventshare__button__box {
    float: right;
}

.events__eventshare__button__text {
    margin: 0;
    padding: 0;
    font-size: 1.45rem;
    margin-left: 0.4rem;
    margin-right: 0.4rem;
    padding-top: 0.6rem;
    padding-bottom: 0rem;
    color: $gray;
}

.events__eventshare__more__button__image {
    width: 1.65rem;
    padding-top: 1.8rem;
}

.events__eventshare__more__button__text {
    margin: 0;
    padding: 0;
    font-size: 1.45rem;
    margin-left: 0.4rem;
    margin-right: 0.4rem;
    margin-top: 1.2rem;
    margin-bottom: 0.8rem;
    color: $gray;
}

.events__eventshare__button__image {
    width: 1.65rem;
    margin-top: 1.2rem;
}

@media all and (-ms-high-contrast:none){
    .events__eventshare__button__image {
        margin-top: 0.8rem;
    }
     *::-ms-backdrop, .events__eventshare__button__image {
        margin-top: 0.8rem;
    }
}

/*----MS Edge Browser CSS Start----*/
@supports ( position:-ms-page ) {
    .events__eventshare__button__image {
        margin-top: 0.8rem;
    }
}
/*----MS Edge Browser CSS End----*/

.events__eventshare__button__image__x {
    width: 1.35rem;
    padding-top: 1.1rem;
    margin-left: 1rem;
}

.events__eventshare__button {
    margin: 0;
    padding: 0;
    display: flex;
    justify-content: center;
    align-content: center;
    background: none;
    border: none;
    position: relative;
    -webkit-transition: all 20ms ease-in;
    -webkit-transform: scale(1); 
    -ms-transition: all 20ms ease-in;
    -ms-transform: scale(1); 
    -moz-transition: all 20ms ease-in;
    -moz-transform: scale(1);
    transition: all 20ms ease-in;
    transform: scale(1);   
}

.events__event__carousel__button__image {
    width: 2.6rem;
    _margin-top: 1.6rem;
    margin-left: 1rem;
}

.events__event__carousel__button__image__fake {
    opacity: 0;
}

.events__event__carousel__button__image__x {
    width: 1.7rem;
    _padding-top: 1.6rem;
    margin-left: 1rem;
}

.events__event__carousel__button {
    margin: 0;
    padding: 0;
    display: flex;
    justify-content: center;
    align-content: center;
    background: none;
    border: none;
    position: relative;
    -webkit-transition: all 20ms ease-in;
    -webkit-transform: scale(1); 
    -ms-transition: all 20ms ease-in;
    -ms-transform: scale(1); 
    -moz-transition: all 20ms ease-in;
    -moz-transform: scale(1);
    transition: all 20ms ease-in;
    transform: scale(1);   
}

.events__eventshare__button:hover {
    text-decoration: none;
    -webkit-transition: all 20ms ease-in;
    -webkit-transform: scale(1.1);
    -ms-transition: all 20ms ease-in;
    -ms-transform: scale(1.1);   
    -moz-transition: all 20ms ease-in;
    -moz-transform: scale(1.1);
    transition: all 20ms ease-in;
    transform: scale(1.1);
}





.event__header__box {
    position: relative;
    display: inline-block;
    text-align: center;
    width: 82rem;
    margin: 0;
    padding: 0;
    padding-top: 0rem;
    padding-bottom: 0.4rem;
}

.event__header__box input {
    font-family: $font;
    font-weight: $fontRegular;
    color: $gray;
    margin: 0;
    padding: 0;
    background: none;
    border: none;
}


.event__header {
    color: $gray;
    font-size: 2.5rem;
    margin: 0;
    padding: 0;
    padding-top: 1rem;
}

.event__header__input {
    font-size: 2.5rem;
    text-align: center;
    margin: 0;
    padding: 0;
    padding-top: 1rem;
}

.event__header__seperator {
    width: 6rem;
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 0.1rem;
    margin-bottom: 0.6rem;
}

.events__seperator__midget {
    margin-bottom: 0;
    opacity: 0;
    filter: alpha(opacity=0);
}

.event__text {
    color: $gray;
    font-size: 1.3rem;
    line-height: 1.3rem;
    margin: 0;
    padding: 0;
}


$eventsEventWidth: 21.3rem;
$eventsEventHeight: 17.5rem;
$eventsEventImageHeight: 14.2rem;


.events__event__box {
    position: relative;
    width: $eventsEventWidth;
    margin-top: 2.2rem;
    margin-left: 1.1rem;
    margin-right: 1rem;
}

.events__event__image {
    width: $eventsEventWidth;
    height: $eventsEventImageHeight;
}

.events__event__title {
    width: $eventsEventWidth;
    color: $white;
    font-size: 1.4rem;
    padding-top: 0.8rem;
    background-color: $green;
    height: 3.3rem;
}

.events__event__add__button {
    width: $eventsEventWidth;
    height: $eventsEventImageHeight;
    margin: 0;
    padding: 0;
    border: none;
    background-color: $green;
}


.events__event__button__box {
    width: $eventsEventWidth;
    height: $eventsEventHeight;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 15;
    overflow: hidden;
    cursor: pointer;
}

.events__event__button {
    background: none;
    border: none;
    width: $eventsEventWidth;
    height: $eventsEventHeight;
    position: absolute;
    _top: $eventsEventHeight;
    top: 0;
    left: 0;
    z-index: 15;
    background-color: rgba(79, 179, 162, 0.68);
    opacity: 0.25;
    -webkit-transition: all $transitionTime ease-in; /* Safari */
    transition: all $transitionTime ease-in;
    margin: 0;
    padding: 0;
}

.events__event__button__box:hover button {
    opacity: 1;
}

.events__event__button__text {
    color: $white;
    font-size: 3rem;
    margin: 0;
    padding: 0;
}

.events__event__button__text__div {
    width: $eventsEventWidth;
    height: $eventsEventHeight;
    background: none;
    position: absolute;
    top: $eventsEventHeight;
    z-index: 16;
    -webkit-transition: all $transitionTime ease-in; /* Safari */
    transition: all $transitionTime ease-in;
}

.events__event__button__box:hover div {
    top: 4.5rem;
}

.events__event__button__image {
    width: 4.5rem;
}



///


.events__events__oneline__header__box {
    text-align: right;
    padding-right: 1rem;
}

.events__events__oneline__more {
    margin: 0;
    padding: 0;
    padding-top: 0.6rem;
    padding-bottom: 0.3rem;
    font-size: 1.5rem;
    color: $gray;
}

.events__events__oneline__header {
    font-size: 1.9rem;
    color: $yellow;
    padding-top: 0.8rem;
}


.events__event__box--event {
    margin-top: 0.7rem;
}



.event__text__box {
    position: relative;
    display: inline-block;
    text-align: center;
    width: 68rem;
    margin: 0;
    padding: 0;
}

.event__text__box input {
    font-family: $font;
    font-weight: $fontRegular;
    position: absolute;
    width: 5rem;
    height: 2rem;
    z-index: 5;
    left: -10%;
    top: 4%;
}

.event__text__box textarea {
    font-family: $font;
    font-weight: $fontRegular;
    display: inline-block;
    text-align: center;
    color: $gray;
    font-size: 1.3rem;
    line-height: 1.6rem;
    width: 68rem;
    margin: 0;
    padding: 0;
    border: none;
    background: none;
    resize: none;
}



.events__event__select__image__icon {
    width: 5rem;
    align-self: center;
}



.events__event__select__image__button__div {
    position: absolute;
    background-color: rgba(83, 176, 161, 0.68);
    border: none;
    display: flex;
    justify-content: center;
    align-items: center;
    opacity: 0;
    -webkit-transition: all $transitionTime ease-in; /* Safari */
    transition: all $transitionTime ease-in;
}

.events__event__select__image__button__div:hover {
    opacity: 1;
}


.events__event__select__image__button__div button {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 50;
    background: none;
    border: none;
}









$eventsEventWidthTablet: 20.55rem;
$eventsEventHeightTablet: 17.5rem;
$eventsEventImageHeightTablet: 14.2rem;
@media (max-width: $tabletMaxWidth) {
    .event__header__box {
        width: 72rem;
    }

    .events__event__box {
        width: $eventsEventWidthTablet;
    }

    .events__event__image {
        width: $eventsEventWidthTablet;
    }

    .events__event__title {
        width: $eventsEventWidthTablet;
    }

    .events__event__add__button {
        width: $eventsEventWidthTablet;
    }


    .events__event__button__box {
        width: $eventsEventWidthTablet;
    }

    .events__event__button {
        width: $eventsEventWidthTablet;
    }

    .events__event__button__text__div {
        width: $eventsEventWidthTablet;
    }
}



$eventsEventWidthMobile: 92vw;
$eventsEventHeightMobile: 86vw;
$eventsEventImageHeightMobile: 60vw;


@media (max-width: $mobileMaxWidth) {

    .events__eventshare__share{
        display: block;
        width: 100%;
        text-align: center;
    }

    .events__eventshare__nav{
        display: flex;
        flex-direction: row-reverse;
        justify-content: center;
        width: 100%;
        margin-bottom: 1rem;
    }

    .events__eventshare__button__box {
        float: none;
        margin-left: 1rem;
        margin-right: 1rem;
        width: 4rem;
    }

    .events__eventshare__button__image {
        width: 4rem;
        margin-top: 1.2rem;
    }

    .events__eventshare__button__image__x {
        width: 2.5rem;
        padding-top: 1.1rem;
        margin-left: 1rem;
    }

    .event__sharestrip__box {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin-top: 1rem;
    }

    .event__header__box {
        width: 100%;
    }

    .event__header__box input {
        margin: 0;
        padding: 0;
    }


    .events__event__box {
        position: relative;
        width: $eventsEventWidthMobile;
        margin-top: 2.2rem;
        margin-left: 1.1rem;
        margin-right: 0rem;
        margin-left: auto;
        margin-right: auto;
    }

    .events__event__image {
        width: $eventsEventWidthMobile;
        height: $eventsEventImageHeightMobile;
    }

    .events__event__title {
        width: $eventsEventWidthMobile;
        font-size: 2.8rem;
        padding-top: 0.8rem;
        height: 5rem;
    }

    .events__event__add__button {
        width: $eventsEventWidthMobile;
        height: $eventsEventHeightMobile;
    }

    .events__event__button__box {
        width: $eventsEventWidthMobile;
        height: $eventsEventHeightMobile;
        margin-left: auto;
        margin-right: auto;
    }

    .events__event__button {
        width: $eventsEventWidthMobile;
        height: $eventsEventHeightMobile;
    }

    .events__event__button__text {
        font-size: 3rem;
        margin: 0;
        padding: 0;
    }

    .events__event__button__text__div {
        opacity: 0;
        top: 0;
    }

    .events__event__button__box button {
        opacity: 0;
        top: 0;
    }

    .events__event__button__box:hover button {
        opacity: 0;
        top: 0;
    }

    .events__event__button__box div {
        opacity: 0;
        top: 0;
    }

    .events__event__button__box:hover div {
        opacity: 0;
        top: 0;
    }

    .event__text__box {
        width: 92%;
    }

    .event__text__box input {
        font-family: $font;
        font-weight: $fontRegular;
        position: absolute;
        width: 5rem;
        height: 2rem;
        z-index: 5;
        left: -10%;
        top: 4%;
    }

    .event__text__box textarea {
        font-size: 3rem;
        line-height: 2.7rem;
        width: 92%;
    }

    .events__eventshare__text {
        font-size: 2.5rem;
        line-height: 2.3rem;
        margin-bottom: 0.5rem;
        display: block;
        width: 100%;
    }

    $iconWidth: 4.7rem;
    $iconHeight: 4.7rem;
    $iconMarginTop: 0.4rem;
    $iconMarginBottom: 0.6rem;

    .events__eventshare__phone__mobile {
        width: $iconWidth;
        height: $iconHeight;
        background-image: url('/images/eventspage/events-sharestrip-phone.svg');
        background-repeat: no-repeat;
        background-size: $iconWidth $iconHeight;
        margin-top: $iconMarginTop;
        margin-bottom: $iconMarginBottom;
        margin-left: 0.8rem;
        margin-right: 0.8rem;
        display: inline-block;
    }

    .events__eventshare__mail__mobile {
        width: $iconWidth;
        height: $iconHeight;
        background-image: url('/images/eventspage/events-sharestrip-mail.svg');
        background-repeat: no-repeat;
        background-size: $iconWidth $iconHeight;
        margin-top: $iconMarginTop;
        margin-bottom: $iconMarginBottom;
        margin-left: 0.8rem;
        margin-right: 0.8rem;
        display: inline-block;
    }

    .events__eventshare__facebook__mobile {
        width: $iconWidth;
        height: $iconHeight;
        background-image: url('/images/eventspage/events-sharestrip-facebook.svg');
        background-repeat: no-repeat;
        background-size: $iconWidth $iconHeight;
        margin-top: $iconMarginTop;
        margin-bottom: $iconMarginBottom;
        margin-left: 0.8rem;
        margin-right: 0.8rem;
        display: inline-block;
    }

    .events__events__oneline__header__box {
        text-align: center;
        width: 100%;
        padding-right: 0rem;
    }

    .events__events__oneline__header {
        font-size: 3rem;
        padding-top: 1rem;
    }

    .events__events__oneline__more {
        padding-top: 0.6rem;
        padding-bottom: 0.5rem;
        font-size: 2.5rem;
    }

    .events__event__box--event {
        margin-top: 0.3rem;
    }
}