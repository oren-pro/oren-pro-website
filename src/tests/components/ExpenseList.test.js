import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render ExpenseList component with expenses correctly', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList component without expenses correctly', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});