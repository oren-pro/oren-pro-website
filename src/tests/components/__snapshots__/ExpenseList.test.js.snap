// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList component with expenses correctly 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id={1}
    key="1"
    notes=""
  />
  <ExpenseListItem
    amount={250}
    createdAt={-345600000}
    description="Ice"
    id={2}
    key="2"
    notes=""
  />
  <ExpenseListItem
    amount={40250}
    createdAt={345600000}
    description="Radioe"
    id={3}
    key="3"
    notes=""
  />
</div>
`;

exports[`should render ExpenseList component without expenses correctly 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
